### PR TITLE
fix types.d.ts + support WebSocket types

### PR DESCRIPTION
**Pull request by kaminskypavel**
the current types definitions is out of sync

### CODE REVIEW

**Comment by balthazar** [comment]
Thanks!

@bennyn Can you take a look?

**Comment by kaminskypavel** [comment]
@balthazar NP, just as a side note, is the "issues" tab disabled on purpose? :)

**Comment by balthazar** [comment]
I think that's how the binance org does its thing, but you can redo it on my original repo, this is just a fork 🙂

**Comment by kaminskypavel** [comment]
its 3 AM here. let me know if you succeeded to merge, or need me to open a PR on your repo.

**Comment by balthazar** [comment]
Merging from here won't be an issue, it's just easier to have the same thing at the same place, and I can't reference the type issue from there in my commit

**Comment by bennyn** [comment]
Let me also have a look at the typings before merging them... Can do that today evening (in ~8 hours).

File: types.d.ts [hunk 1]
```diff
@@ -1,79 +1,204 @@
+// Built With ♡ 'PK' (www.pavel-kaminsky.com)
```
Comment (bennyn) [inline types.d.ts]: We shouldn't advertise in code.

File: types.d.ts [hunk 2]
```diff
@@ -1,79 +1,204 @@
+// Built With ♡ 'PK' (www.pavel-kaminsky.com)
+
+// tslint:disable:interface-name
+import {WebSocket} from "binance-api-node";
```
Comment (bennyn) [inline types.d.ts]: Why did you import the `WebSocket` interface?

File: types.d.ts [hunk 3]
```diff
@@ -1,79 +1,204 @@
...
+        | 'MARKET'
+        | 'STOP_LOSS'
+        | 'STOP_LOSS_LIMIT'
+        | 'TAKE_PROFIT'
+        | 'TAKE_PROFIT_LIMIT';
+
+    export type TimeInForce = 'GTC' | 'IOC';
+}
+
+declare module Results {
```
Comment (bennyn) [inline types.d.ts]: Why do you declare a new module? Can't we put the interfaces inside of that also in the main `'binance-api-node'` module?

File: types.d.ts [hunk 4]
```diff
@@ -1,79 +1,204 @@
...
+
+    export interface AssetBalance {
+        asset: string;
+        free: string;
+        locked: string;
+    }
+
+    export interface Binance {
+        accountInfo(): Promise<Account>;
+
```
Comment (bennyn) [inline types.d.ts]: For consistency reasons, empty lines within `Binance` interface should be removed.

**Comment by bennyn** [comment]
I reviewed the code, thanks for your submission @kaminskypavel. It's great that you help improving type definitions. 

@balthazar Very good thing is that this PR doesn't break the current typings that we have in place, which means that applications built with TypeScript, which are using "binance-api-node", will still continue to work. 🎉 

@kaminskypavel: Do you know how we can achieve getting a typed client? Because at the moment you would need to do the following:

```ts
import Binance from 'binance-api-node'

const client: any = Binance({
  apiKey: process.env.BINANCE_API_KEY,
  apiSecret: process.env.BINANCE_API_SECRET,
})
```

I don't like that `client` is of type `any` in this case. I would rather prefer something like:

```ts
import Binance from 'binance-api-node'

const client: Binance = Binance({
  apiKey: process.env.BINANCE_API_KEY,
  apiSecret: process.env.BINANCE_API_SECRET,
})
```

Do you have a solution to this issue?

File: types.d.ts (hunk 1 above)
Comment (kaminskypavel) [reply types.d.ts]: whoops.. that slipped in. will remove

File: types.d.ts (hunk 3 above)
Comment (kaminskypavel) [reply types.d.ts]: we totally can, i didnt want to pollute the main module.
would you prefer that?

**Comment by kaminskypavel** [comment]
@bennyn thanks for the review.
regarding your question, client is not of type "any". 
here have a look :

image

File: types.d.ts (hunk 3 above)
Comment (bennyn) [reply types.d.ts]: Yipp, I would prefer having it in the main module.

**Comment by bennyn** [comment]
> @bennyn thanks for the review.
regarding your question, client is not of type "any".
here have a look :

Cool, are you also using "WebStorm"? 😃 Which view did you open to get that inspection?

**Comment by kaminskypavel** [comment]
@bennyn , just press Ctrl+q (quick docs).

File: types.d.ts [hunk 5]
```diff
@@ -1,79 +1,187 @@
+// tslint:disable:interface-name
```
Comment (bennyn) [inline types.d.ts]: There is no TSLint in this project so we don't need an instruction for it except there is a case where it breaks code for people using TSLint and our library. If `// tslint:disable:interface-name` gets removed, will TSLint complain when "binance-api-node" is used as a dependency?

**Comment by bennyn** [comment]
> @bennyn , just press Ctrl+q (quick docs).

+1 for using TypeScript, +1 for using WebStorm, +1 for using Windows and another +1 for making this PR 😄 

I just commented on the `tslint:disable` declaration. If we sort this out, then the PR is ready to go in. 👏

**Comment by kaminskypavel** [comment]
that's alot of +1s , thanks 😄
We can resolve this by refactoring all the interfaces to start with a capital "I". 
but I suggest we keep it for another PR as this will be a breaking change.

**Comment by bennyn** [comment]
Ok!

**Pull request merged**